etc/migration/12.1-to-12.2/2.tenantEmailDomains.js

```diff
@@ -0,0 +1,127 @@
+ *  - Drop the `emailDomain` value
+ */
+
+var _ = require('underscore');
+var optimist = require('optimist');
+var path = require('path');
+
+var Cassandra = require('oae-util/lib/cassandra');
+var log = require('oae-logger').logger('tenants-email-domains-migrator');
+var OAE = require('oae-util/lib/oae');
+var TenantsAPI = require('oae-tenants');
+
+var argv = optimist.usage('$0 [--config <path/to/config.js>] [--warnings <path/to/warnings.csv>]')
+    .alias('c', 'config')
+    .describe('c', 'Specify an alternate config file')
+    .default('c', 'config.js')
+
+    .alias('w', 'warnings')
+    .describe('w', 'Specify the path to the file where unmappable users should be dumped to')
+    .default('w', 'email-migration.csv')
+
+    .alias('h', 'help')
+    .describe('h', 'Show usage information')
+    .argv;
+
+if (argv.help) {
+    optimist.showHelp();
+    process.exit(0);
+}
+
+// Get the config
+var configPath = path.resolve(process.cwd(), argv.config);
+var config = require(configPath).config;
+
+// Ensure that this application server does NOT start processing any preview images
+config.previews.enabled = false;
+
+// Ensure that we're logging to standard out/err
+config.log = {
+    'streams': [
+        {
+            'level': 'info',
+            'stream': process.stdout
+        }
+    ]
+};
+
+// Spin up the application container. This will allow us to re-use existing APIs
+OAE.init(config, function(err) {
+    if (err) {
+        log().error({'err': err}, 'Unable to spin up the application server');
+        return process.exit(err.code);
+    }
+
+    // Get all the tenants
+    Cassandra.runQuery('SELECT "alias", "emailDomain" FROM "Tenant"', [], function(err, rows) {
+        if (err) {
+            log().error({'err': err}, 'Unable to get all the tenants from the database');
+            return process.exit(err.code);
+        }
+
+        // Iterate through all the tenants and copy the emailDomain
+        var tenantsToUpdate = _.chain(rows)
+            .map(Cassandra.rowToHash)
+            .filter(function(hash) {
+                return (hash.emailDomain)
+            })
+            .value();
+
+        // Update all the tenants
+        _updateTenants(tenantsToUpdate, function(err) {
+            if (err) {
+                log().error({'err': err}, 'Unable to copy the emailDomain value');
+                return process.exit(err.code);
+            }
+
+            log().info('Updated all tenants');
+            process.exit(0);
+        });
+    });
+});
+
+/**
+ * Update a set of tenants their email domains
+ *
+ * @param  {Object[]}   tenants         Each object represent a tenant that needs to be updated. The `alias` and `emailDomain` keys should be present
+ * @param  {Function}   callback        Standard callback function
+ * @param  {Object}     callback.err    An error that occurred, if any
+ * @api private
+ */
+var _updateTenants = function(tenants, callback) {
```

Comment:
Since we're running in a scripted environment, can you make this `function _updateTenants` just to avoid hoisting-related oddities